tcpClientServer/tests/tst_tcpclientserver.cpp

```diff
@@ -40,6 +42,27 @@ void tcpClientServer::clientFailedInitTest() {
     clientProcess.waitForFinished();
     QCOMPARE(clientProcess.exitCode(), QProcess::CrashExit);
 }
+
+void tcpClientServer::succeededInitTest() {
+    QProcess serverProcess;
+    serverProcess.start(QString("../tcp-server"), QStringList());
+    serverProcess.waitForStarted();
+
+    QTest::qSleep(1000);
+
+    QProcess clientProcess;
+    clientProcess.start(QString("../tcp-client"), {"--only-init"});
+    clientProcess.waitForStarted();
```

Comment:
Qui forse userei la `waitForReadyRead` piuttosto che la `waitForStarted`, visto che la prima implica sia la seconda, sia che il processo è pronto per leggere il nostro input:
```suggestion
    clientProcess.waitForReadyRead();
```